### DISCUSSION TITLE
Populate Repeater

### DISCUSSION THREAD

**Discussion by ismaeltj9** [filamentphp/filament#10247] [Help] [answered]
Hello all, I am trying to populated a repeater with some data, in the future I will populate with a collection of model, is this possible? for now I am trying to populate like this but the value is not filling

`                        ->fillForm(
                            [
                                'aaa' => [
                                    'value' => ['teste1'], ['teste2'], ['teste3'], ['teste4'], ['teste5']
                                ],
                            ])` 
`                         ->form([
                            Forms\Components\Repeater::make('aaa')
                                ->schema([
                                    Forms\Components\TextInput::make('value')
                                ]),
                        ])`

![image](https://github.com/filamentphp/filament/assets/24549114/28de1d5c-3742-47b1-a66f-33e4ce34977b)

**Comment by leandrocfe** [comment]
```php
Repeater::make('repeater')
    ->schema([
        TextInput::make('value'),
    ])
    ->default([
        ['value' => '1'],
        ['value' => '2'],
        ['value' => '3'],
    ])
```
?

**Reply by ismaeltj9** [reply]
Olá @leandrocfe , obrigado pela resposta, o problema é que eu tenho uma bulkaction, dentro dela eu preciso filtrar alguns registros do que ta selecionado ($records->where(....)) e gerar um form com varios repeater preenchidos com esses registros filtrados, tem algum recurso fácil pra fazer isso?

**Comment by ismaeltj9** [comment, marked as answer]
Eu fiz desse jeito, nem sabia que era possivel:

`                    Tables\Actions\BulkAction::make('Materiais3')
                        ->form(function (Collection $records) {
                            $f = [];
                            foreach ($records as $material) {
                                $f[] = Forms\Components\Select::make($material->nome)
                                    ->options(
                                        Fornecedor::find($material->itens->pluck('fornecedor_id'))->pluck('nome')
                                    );
                            }
                            return $f;
                        })`

**Reply by mitsyara** [reply]
where's the part you filled the repeater with records data?
